scripts/SANS/sans/gui_logic/models/table_model.py

```diff
@@ -128,6 +128,14 @@ def create_empty_row():
         row = [''] * 16
         return TableIndexModel(*row)
 
+    def get_non_empty_rows(self, rows):
+        non_empty_rows = []
+        for index in rows:
+            row = self.get_table_entry(index)
+            if not row.is_empty():
+                non_empty_rows.append(index)
+        return non_empty_rows
```

Comment:
Could you use a list comprehension here?
`non_empty_rows = [index for index in rows if not self.get_table_entry(index).is_empty()]`

Reply:
I agree this is unnecessarily terse. I've changed it to use filter (filter is generally faster than list comprehensions)